### ISSUE TITLE
Make history accessible from BS as it is in App

### ISSUE THREAD

**Issue by AbeFM** [bambulab/BambuStudio#593]
Often I have to use my phone app to reprint something. It would be great if BS could access my history as smoothly as the phone does.

Similarly, if there was a way to find every .3mf I've saved from inside BS instead of having to look though my file system, that would be awesome.

**Comment by YifanWuBambu** [comment]
For the print history, we have added it to our plan list.
For local file history, you can find it at home page.
![image](https://user-images.githubusercontent.com/110661856/200825920-2ba61c56-e72f-4257-8198-6ec99e740652.png)

**Comment by AbeFM** [comment]
Yeah, that's it - I want that, but displaying the same history I would get on the phone.

The local list seems short, why not have it be very long? I'd give up a MB of disk space to be able to find a file I printed a few months ago.

Renaming the history would be good, too. I have a lot of things in my history that have the name of something I'd done earlier.

**Comment by YifanWuBambu** [comment]
The history is not the same as the phone.

The phone shows printing history, but Bambu Studio shows editing history.
They are not the same.

I have put the requirement to our plan.

**Comment by le-thanh** [comment]
I have an addition to this feature.  I need to export the print history to a CSV which I could then import to a spreadsheet or database.

I run a small business and could use this info to help with analytics, such as material, printing costs, print time, success rate etc.  I currently do this manually, having to scroll through the history in the BS Handy app and then perform manual data entry to 'transfer' this info so I can link to a set of data for my filament inventory so I can track how much has been used and what is remaining.

being able to export the history would save a huge amount of time (plus manual tasks)

Also, while not part of this topic, but related to my workflow, being able to allow the ams to read generic NFC tags, or allow us to buy AMS-compatible NFC tags to put on our spools would be very useful
